test/pullrequest/e2e_test.go

```diff
@@ -0,0 +1,170 @@
+package pullrequest
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/jenkins-x/go-scm/scm"
+)
+
+var (
+	proxy = flag.Bool("use-fake-scm-proxy", true, "whether to use fake in-memory proxy instead of using real SCM servers")
+)
+
+func TestPullRequest(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", t.Name())
+	if err != nil {
+		log.Fatalf("ioutil.TempDir: %v", err)
+	}
+	t.Logf("Test dir: %s", tmpdir)
+	defer os.RemoveAll(tmpdir)
+
+	// Make sure command is installed locally to resolve any dependencies.
+	// We need to do this first since the run might try to pull in new
+	// dependencies through the proxy.
+	install := exec.Command("go", "get", "-u", "github.com/tektoncd/pipeline/cmd/pullrequest-init")
```

Comment:
im confused by why this and the below `go run` are needed - could you explain a bit more? i thought the proxy was just receiving HTTP requests and replying, not sure why extra dependencies are needed?